azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -224,6 +218,226 @@ def create(
+        endpoint_type: str,
+        endpoint_account_name: Optional[str] = None,
+        endpoint_resource_group: Optional[str] = None,
+        endpoint_subscription_id: Optional[str] = None,
+        endpoint_policy_name: Optional[str] = None,
+        connection_string: Optional[str] = None,
+        container_name: Optional[str] = None,
+        encoding: Optional[str] = None,
+        batch_frequency: Optional[int] = None,
+        chunk_size_window: Optional[int] = None,
+        file_name_format: Optional[str] = None,
+        endpoint_uri: Optional[str] = None,
+        entity_path: Optional[str] = None,
+        database_name: Optional[str] = None,
+        primary_key: Optional[str] = None,
+        secondary_key: Optional[str] = None,
+        partition_key_name: Optional[str] = None,
+        partition_key_template: Optional[str] = None,
+        identity: Optional[str] = None
+    ):
+        # if nothing is provided -> should we block?
+        # have the user say the type. Will make args easier (as in we do not need to check for unneeded args)
+        original_endpoint = self._show_by_type(endpoint_name=endpoint_name, endpoint_type=endpoint_type)
+
+        if any([connection_string, primary_key, secondary_key]) and identity:
+            optional_msg = ", --primary-key and/or --secondary-key," if primary_key or secondary_key else ""
+            error_msg = "Please use either --connection-string" + optional_msg + " or --identity."
+            raise MutuallyExclusiveArgumentError(error_msg)
+
+        # Properties for all endpoint types
+        if endpoint_resource_group:
+            original_endpoint.resource_group = endpoint_resource_group
+        if endpoint_subscription_id:
+            original_endpoint.subscription_id = endpoint_subscription_id
+        if endpoint_uri:
+            # Handle this later with cosmos db connection string parsing
+            original_endpoint.endpoint_uri = endpoint_uri
+
+        # Identity/Connection String schenanigans
+        # If Identity and Connection String args are provided, Identity wins
+        fetch_connection_string = False
+        if identity:
+            if EndpointType.CosmosDBContainer.value == endpoint_type.lower():
+                original_endpoint.primary_key = None
+                original_endpoint.secondary_key = None
+            else:
+                original_endpoint.connection_string = None
+            original_endpoint.authentication_type = AuthenticationType.IdentityBased.value
+            if identity == SYSTEM_ASSIGNED_IDENTITY:
+                original_endpoint.identity = None
+            else:
+                original_endpoint.identity = ManagedIdentity(
+                    user_assigned_identity=identity
+                )
+        elif any([connection_string, primary_key, secondary_key]):
+            original_endpoint.identity = None
+            original_endpoint.authentication_type = AuthenticationType.KeyBased.value
```

Comment:
In that case, does it make more sense to set it to `None` after `if identity` before we start doing checks, and only set it if we need to set it to identity-based? 

Otherwise it's probably fine, I remember a similar issue cropping up in CLI core - as long as it's correctly set to identity when we need it to be, defaulting to keybased is probably fine as it's been the de-facto default since the beginning of the feature I believe.

Reply:
I am assuming you are saying set auth type to None in this elif statement. Which is reasonable. But will keep it as keybased so it is more consistent with create